state-transition/core/state_processor_withdrawals.go

```diff
@@ -136,3 +155,194 @@ func (sp *StateProcessor) processWithdrawals(
 
 	return nil
 }
+
+// processWithdrawalRequest is the equivalent of process_withdrawal_request as defined in the spec.
+// It should only be called after the electra hard fork.
+// For invalid withdrawal requests, we return nil, and only return error for system errors.
+func (sp *StateProcessor) processWithdrawalRequest(st *state.StateDB, withdrawalRequest *ctypes.WithdrawalRequest) error {
+	amount := withdrawalRequest.Amount
+	// If the amount is 0, it's a full exit.
+	isFullExitRequest := amount == constants.FullExitRequestAmount
+	pendingPartialWithdrawals, err := st.GetPendingPartialWithdrawals()
+	if err != nil {
+		return err
+	}
+	// If partial withdrawal queue is full, only full exits are processed
+	if len(pendingPartialWithdrawals) == constants.PendingPartialWithdrawalsLimit && !isFullExitRequest {
+		sp.logger.Warn(
+			"skipping processing of withdrawal request as partial withdrawal queue is full",
+			withdrawalFields(withdrawalRequest, nil)...,
+		)
+		return nil
+	}
+	index, validator, err := validateWithdrawal(st, withdrawalRequest)
+	if err != nil {
+		sp.logger.Info("Failed to validate withdrawal", withdrawalFields(withdrawalRequest, err)...)
+		// Note that we do not return error on invalid requests as it's a user error and invalid withdrawal requests are simply skipped.
+		// Since we're interacting with the KV Store, i/o errors could occur, but we expect these to be sparse enough
+		// that it won't impact consensus. We could consider returning error on codec.ErrEncoding rather than silencing here.
```

Comment:
I would drop this part of the comment. In order for that to happen we need a super majority of nodes to have data corruption for the very same block. I don't think we need to account for that (and a request can always be re-issued)